src/event-bridge/event-bus.ts

```diff
@@ -40,7 +40,12 @@ export const isEventBus = <E extends Event>(v: any): v is IEventBus<E> => {
   );
 };
 
-export interface IEventBusFilterable<E extends Event> {
+/**
+ * Returns the {@link Event} type on the {@link EventBus}.
+ */
+export type EventBusEvent<B extends IEventBus<any>> = [B] extends [IEventBus<infer E>] ? E : never;
```

Comment:
Should this be called GetEventBusType? Or else it could be confused with the Event type. The verb helps with understanding